### PR TITLE
[Move] Implement pollen puff

### PR DESCRIPTION
**Pull request by prateau**
## What are the changes?
Implementing the move Pollen Puff. There was already this PR on this subject but it was incomplete and closed due to inactivity.

## Why am I doing these changes?
Unimplemented move

## What did change?
Pollen puff effect change when used on ally : the ally is now restored 50% of max health instead of taking damage.
Pokemon with Bulletproof or Telepathy are not affected by the heal effect.
As Heal Block is unimplemented, the interraction of pollen puff with heal block isn't implemented.

### Screenshots/Videos
Pollen puff damaging opponent and healing ally
https://github.com/pagefaultgames/pokerogue/assets/56865723/0f31df5d-44c2-4595-b80a-7b08590f3c73

Pollen puff is innefective on pokemon with Bulletproof
https://github.com/pagefaultgames/pokerogue/assets/56865723/4e8478fe-56e9-41d4-90df-fbed62b45239

Pollen puff is innefective on ally with Telepathy
https://github.com/pagefaultgames/pokerogue/assets/56865723/645436d8-f455-4a96-af15-35f2891dfed7

## How to test the changes?
You can modify src/overrides.ts to test the move in a double battle with the different abilities

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [x] Are the changes visual?
  - [x] Have I provided screenshots/videos of the changes?

### CODE REVIEW

File: src/data/move.ts [hunk 1]
```diff
@@ -3043,6 +3066,31 @@ export class TeraBlastCategoryAttr extends VariableMoveCategoryAttr {
   }
 }
 
+/**
+ * Change the move category to status when used on the ally
+ * @extends VariableMoveCategoryAttr
+ * @see {@linkcode apply}
+ */
+export class StatusCategoryOnAllyAttr extends VariableMoveCategoryAttr {
```
Comment (Tempo-anon) [inline src/data/move.ts:3074]: Can we get a test with this such as the ally having good as gold?

**Review by Tempo-anon — APPROVED** [review]
Looks pretty good to me. Just one edge case I want to confirm

File: src/data/move.ts (hunk 1 above)
Comment (prateau) [reply src/data/move.ts:3074]: The ally pokemon with good as gold is affected by the move. This is the intended behavior for pollen puff at least (based on my test on Showdown)

https://github.com/pagefaultgames/pokerogue/assets/56865723/0c9ddbd4-9d50-447d-8229-bc953813c616
